fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -234,24 +234,25 @@ public static Object toClassWithRealEquals(@Nullable Object obj) {
 
     @SuppressWarnings("unchecked")
     public static int compare(@Nullable Object fieldValue, @Nullable Object comparand) {
-        if (fieldValue == null) {
-            if (comparand == null) {
-                return 0;
-            } else {
-                return -1;
-            }
-        } else if (comparand == null) {
-            return 1;
+        return toComparable(fieldValue).compareTo(toComparable(comparand));
```

Comment:
I think the original line 237~244 is wrong? If any of the left or right is null, the comparison result should be unknown? Also in the original code, we call: `if(value == null) return null` before we call this `compare` method, so the block of `if(fieldValue == null)` was never reached.

Reply:
Got it, I changed the method to compare between non-null values.